tests/lib/edge_impulse/src/main.cpp

```diff
@@ -434,8 +446,11 @@ ZTEST(suite0, test_data_isr)
 static void setup_fn(void *unused)
 {
 	ARG_UNUSED(unused);
+
 	bool cancelled;
 	int err = ei_wrapper_clear_data(&cancelled);
+	prediction_idx = 0;
+	ei_run_classifier_mock_init();
```

Comment:
Please take the semaphore here to ensure we do not start test with already given semaphore (it would make unit tests more robust)